src/onnx/parse_quantizelinear.cpp

```diff
@@ -37,47 +38,74 @@ struct parse_quantizelinear : op_parser<parse_quantizelinear>
+        }
+
+        // Starting with version 19 ONNX introduced the constraint that x and y_scale types must be
+        // the same
+        if(parser.opset_version >= 19 and
+           args[0]->get_shape().type() != args[1]->get_shape().type())
+        {
+            MIGRAPHX_THROW("QuantizeLinear: x and y_scale must be of same type");
+        }
+
+        if(args.size() == 3 and args[1]->get_shape().lens() != args[2]->get_shape().lens())
+        {
+            MIGRAPHX_THROW(
+                "QuantizeLinear: y_scale and y_zero_point shapes must be equal. Provided y_scale "
+                "shape: " +
+                to_string_range(args[1]->get_shape().lens()) +
+                ", provided y_zero_point shape: " + to_string_range(args[2]->get_shape().lens()));
+        }
+
         int axis = 1;
         if(contains(info.attributes, "axis"))
             axis = info.attributes.at("axis").i();
 
-        auto input_lens = args[0]->get_shape().lens();
-        auto n_dim      = input_lens.size();
+        int block_size = 0;
+        if(contains(info.attributes, "block_size"))
+            block_size = info.attributes.at("block_size").i();
 
-        instruction_ref y_scale = args[1];
-        if(args[1]->get_shape().elements() != 1)
+        std::optional<migraphx::shape::type_t> output_type;
+        if(contains(info.attributes, "output_dtype"))
         {
-            auto tuned_axis = tune_axis(n_dim, axis, opd.op_name);
-            y_scale         = info.add_instruction(
-                make_op("broadcast", {{"axis", tuned_axis}, {"out_lens", input_lens}}), args[1]);
+            output_type = get_type(info.attributes.at("output_dtype").i());
         }
 
-        auto common_args = add_common_args(*info.mod, {args[0], y_scale});
-
-        if(args.size() == 3)
+        if(output_type.has_value() and args.size() == 3 and
+           *output_type != args[2]->get_shape().type())
         {
-            auto y_zero_point = args[2];
-            if(y_zero_point->get_shape().elements() != 1)
-            {
-                auto tuned_axis = tune_axis(n_dim, axis, opd.op_name);
-                y_zero_point    = info.add_instruction(
-                    make_op("broadcast", {{"axis", tuned_axis}, {"out_lens", input_lens}}),
-                    y_zero_point);
-            }
-            else
-            {
-                y_zero_point = info.add_instruction(
-                    make_op("multibroadcast", {{"out_lens", input_lens}}), y_zero_point);
-            }
+            MIGRAPHX_THROW(
+                "QuantizeLinear: output_type and y_zero_point type must match. output_type: " +
+                to_string(*output_type) +
+                +", y_zero_point type: " + to_string(args[2]->get_shape().type()));
+        }
+
+        transform_quantize_dequantize_linear_inputs(info, opd.op_name, block_size, axis, args);
 
-            common_args.push_back(y_zero_point);
+        if(parser.opset_version < 19)
+        {
```

Comment:
I thought about adding these as well, but decided against mainly because I haven't noticed that it's common practice to have the type constraints checked in parser code, although I might be wrong here.